app/authentication.py

```diff
@@ -17,9 +17,13 @@ def token_is_valid(incoming_token, module):
 
 def get_allowed_tokens_from_config(config, module='main'):
     """Return a list of allowed auth tokens from the application config"""
-    env_variable_name = 'DM_ANTIVIRUS_API_AUTH_TOKENS'
+    env_variable_name = ""
 
-    if module == 'callbacks':
+    if module == "main":
+        env_variable_name = 'DM_ANTIVIRUS_API_AUTH_TOKENS'
+    elif module == 'callbacks':
+        # though SNS uses the "Basic" auth scheme, we can treat the payload section of the header as a single opaque
+        # token set in DM_ANTIVIRUS_API_CALLBACK_AUTH_TOKENS like any normal "Bearer" tokens.
```

Comment:
So when SNS uses Basic auth, we're not actually decoding the base64 and splitting it up, like with the Bearer token. We're just taking the base64 encoding as is, and comparing it with our known value from the config?

Reply:
:grin: